browser-extension/mv2/src/background/background.js

```diff
@@ -1153,22 +1184,21 @@ BG.Methods.handleClientPortConnections = () => {
 
     const tabId = senderTab.id;
 
-    // documentLifeCycle is only used by chrome and not firefox
-    if (!port.sender.documentLifecycle || port.sender.documentLifecycle === "active") {
-      window.tabService.resetPageData(senderTab.id);
-      window.tabService.setData(tabId, BG.TAB_SERVICE_DATA.CLIENT_PORT, port);
+    window.tabService.resetPageData(senderTab.id);
+    window.tabService.setData(tabId, BG.TAB_SERVICE_DATA.CLIENT_PORT, port);
```

Comment:
Pagedata will be always set irrespective of whether page is in active state or prerendered state. 
When the tab is commited, the state is updated to the correct state